sympy/physics/quantum/tests/test_grover.py

```diff
@@ -4,16 +4,16 @@
 from sympy.physics.quantum.qapply import qapply
 from sympy.physics.quantum.qubit import IntQubit
 from sympy.physics.quantum.grover import (apply_grover, superposition_basis,
-        OracleGate, grover_iteration, WGate)
+        OracleGate, grover_iteration, WGate, OracleGateFunction)
 
 
 def return_one_on_two(qubits):
     return qubits == IntQubit(2, qubits.nqubits)
-
+return_one_on_two=OracleGateFunction(return_one_on_two)
 
 def return_one_on_one(qubits):
     return qubits == IntQubit(1, nqubits=qubits.nqubits)
-
+return_one_on_one=OracleGateFunction(return_one_on_one)
```

Comment:
I guess you could use this as a decorator like:
```
@OracleGateFunction()
def return_one_on_one(qubits):
    ...
```

Reply:
Good idea, I'll add such an example to the docstring too, since it is potentially a lot more convincing for a user to use it in that way.